static/src/js/components/Table/Table.jsx

```diff
@@ -15,12 +22,139 @@ import For from '../For/For'
+
+  // Current page of rows
+  const pagedRows = data ? data.slice(startIndex, endIndex) : []
+
+  const lastPageNum = parseInt(Math.ceil(rowCount / defaultPageSize), 10)
+
+  // // Does this provider have enough Rows to page? We hide the pagination buttons if not
+  const hasPages = rowCount > defaultPageSize
+
+  const defaultPaginationStyles = {
+    minWidth: '2.5rem',
+    textAlign: 'center'
+  }
+
+  useEffect(() => {
+    if (!loading) {
+      setRowCount(data.length)
+    }
+  }, [loading, data])
+
   const renderHeaders = headers.map((header) => (
     <th key={header}>{header}</th>
   ))
 
+  const handleItemClick = (currentPage) => {
+    setPageNum(currentPage)
+    setCurrentCount(currentPage * defaultPageSize)
+  }
+
+  const generatePaginationItems = () => {
+    // Only show 3 pages, the current page and one before or after (within the valid range of pages)
+    const pages = [pageNum - 1, pageNum, pageNum + 1]
+      .filter((page) => page >= 1 && page <= lastPageNum)
+
+    const returnItems = []
+
+    // If the first page is not 1, add the pagination item for page 1
+    if (pages[0] !== 1) {
+      returnItems.push(
+        <Pagination.Item
+          key="page-1"
+          onClick={() => handleItemClick(1)}
+          active={pageNum === 1}
+          style={defaultPaginationStyles}
+        >
+          {1}
+        </Pagination.Item>
+      )
+
+      // And if the first page is not 2, add an ellipsis
+      if (pages[0] !== 2) {
+        returnItems.push(
+          <Pagination.Ellipsis
+            key="page-ellipsis-1"
+            disabled
+          />
+        )
+      }
+    }
+
+    pages.forEach((page) => {
+      returnItems.push(
+        <Pagination.Item
+          key={`page-${page}`}
+          onClick={() => handleItemClick(page)}
+          active={page === pageNum}
+          style={defaultPaginationStyles}
+        >
+          {page}
+        </Pagination.Item>
+      )
+    })
+
+    // If the last page is not lastPageNum, add the pagination item for the lastPageNum
+    if (pages[pages.length - 1] !== lastPageNum) {
+      // And if the last page is not lastPageNum - 1, add an ellipsis
+      if (pages[pages.length - 1] !== lastPageNum - 1) {
+        returnItems.push(
+          <Pagination.Ellipsis
+            key="page-ellipsis-2"
+            disabled
+          />
+        )
+      }
+
+      returnItems.push(
+        <Pagination.Item
+          key={`page-${lastPageNum}`}
+          onClick={() => handleItemClick(lastPageNum)}
+          active={pageNum === lastPageNum}
+          style={defaultPaginationStyles}
+        >
+          {lastPageNum}
+        </Pagination.Item>
+      )
+    }
+
+    return returnItems
+  }
+
+  const handlePageChange = (direction) => {
+    const newPageNum = pageNum + direction
+    const newCurrentCount = currentCount + direction * defaultPageSize
+
+    setPageNum(newPageNum)
+    setCurrentCount(newCurrentCount)
+  }
+
+  const pagination = (
```

Comment:
I feel like this class is getting pretty long, wonder if it would be better to pull it out into a separate component?

Reply:
Agreed! I had a comment about breaking it up.

Reply:
Please let me know ASAP if it is necessary for an MVP. I can add this to the ticket I'll be creating today to make limit and offset available.